repo/blob/filesystem/filesystem_storage.go

```diff
@@ -231,7 +230,19 @@ func (fs *fsImpl) ReadDir(ctx context.Context, dirname string) ([]os.FileInfo, e
 		return nil, err
 	}
 
-	return v.([]os.FileInfo), nil
+	fileInfos := make([]os.FileInfo, 0, len(v.([]os.DirEntry)))
+
+	for _, e := range v.([]os.DirEntry) {
+		fi, err := e.Info()
+		if err != nil {
+			// nolint:wrapcheck
+			return nil, err
+		}
+
+		fileInfos = append(fileInfos, fi)
+	}
+
+	return fileInfos, nil
```

Comment:
This is an easier workaround to return the results as `[]os.FileInfo`. Changing the method signature from `ReadDir(ctx context.Context, path string) ([]os.FileInfo, error)` to `ReadDir(ctx context.Context, path string) ([]os.DirEntry, error)` will require massive refactoring of other storage implementations (`davStorageImpl`, `fsImpl`, and `sftpImpl`) as well.